lms/djangoapps/courseware/views.py

```diff
@@ -1370,3 +1371,33 @@ def _track_successful_certificate_generation(user_id, course_id):  # pylint: dis
                 }
             }
         )
+
+
+@require_http_methods(["GET", "POST"])
+def render_xblock(request, usage_key_string):
+    """
+    Returns an HttpResponse with HTML content for the xBlock with the given usage_key.
+    The returned HTML is a chromeless rendering of the xBlock (excluding content of the containing courseware).
+    """
+    usage_key = UsageKey.from_string(usage_key_string)
+    usage_key = usage_key.replace(course_key=modulestore().fill_in_run(usage_key.course_key))
+    course_key = usage_key.course_key
```

Comment:
I had verified with @doctoryes that the `UsageKey` (on its own) is sufficient as a unique identifier for a block.  The issue you raise is relevant to only old Mongo courses, which are now deprecated.  For Split-based courses, the `CourseKey` embedded in the `UsageKey` contains the run so there would be no disambiguity.
And, because of the limitation in old Mongo, we shouldn't have any courses in the old Mongo modulestore that differ by only run.  The `create_course` method in old Mongo prevents creating a duplicate course that differs by only its run.

@doctoryes Please correct me if I'm incorrect.